### PR TITLE
Add documentation for http_srv example

### PR DESCRIPTION
**Pull request by davidkroell**
This is the first version of the documentation for the HTTP Server added in #35.
Please review and give feedback for further improvement.

Best regards,
David

### CODE REVIEW

File: http_srv.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+**Create a new person**
+```bash
+curl -X POST http://localhost:8080/people/create \
+    -H "Content-Type: application/x-www-form-urlencoded" \
+    -d 'first_name=John&last_name=Doe'
+
+### Output: created
+```
+When this command is executed, a new person gets stored in the `people` map.
+The attributes fo the person, namely first- and lastname, are specified in the POST body
```
Comment (sagar-jadhav) [inline http_srv.md]: minor typo it should be `firstname` instead of  `first-`

File: http_srv.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+URL-encoded. In this example `John Doe` is used.
```
Comment (sagar-jadhav) [inline http_srv.md:26]: Add `createPerson` code snippet here as follows:

```
func createPerson(w http.ResponseWriter, r *http.Request) {
	if r.Method == "POST" {

		fn := r.FormValue("first_name")
		ln := r.FormValue("last_name")

		p := Person{
			FirstName: fn,
			LastName:  ln,
		}

		people[lastId] = p
		lastId++

		w.WriteHeader(http.StatusCreated)
		w.Write([]byte("created"))

	} else {
		w.WriteHeader(http.StatusBadRequest)
		w.Write([]byte("bad request"))
	}
}
```

File: http_srv.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+**List all people**
+```bash
+curl http://localhost:8080/people
+
+### Output: {"0":{"first_name":"John","last_name":"Doe"}}
+```
+This command calls the `readPeople()` function.
+It returns all the people currently stored in the data map in a JSON format.
+
```
Comment (sagar-jadhav) [inline http_srv.md:62]: Add `readPeople` code snippet as follows:

```
func readPeople(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)

	err := json.NewEncoder(w).Encode(people)
	if err != nil {
		log.Println(err)
	}
}
```

File: http_srv.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+For the first person inserted, this is (of course) 0.
+
+```bash
+curl -X PUT http://localhost:8080/people/update?id=0 \
+     -H "Content-Type: application/x-www-form-urlencoded" \
+     -d 'first_name=Alice&last_name=Smith'
+
+### Output: updated
+```
+
```
Comment (sagar-jadhav) [inline http_srv.md:89]: Add `updatePerson` code snippet here as follows:

```
func updatePerson(w http.ResponseWriter, r *http.Request) {
	id := r.FormValue("id")
	intId, err := strconv.Atoi(id)
	if err != nil {
		w.WriteHeader(http.StatusBadRequest)
		w.Write([]byte("bad request"))
		return
	}

	person, ok := people[intId]
	if !ok {
		w.WriteHeader(http.StatusNotFound)
		w.Write([]byte("not found"))
		return
	}

	person.FirstName = r.FormValue("first_name")
	person.LastName = r.FormValue("last_name")

	people[intId] = person
	w.WriteHeader(http.StatusAccepted)
	w.Write([]byte("updated"))
}
```

File: http_srv.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+**Delete a person**
+
+Deleting the only person in the map by calling the following HTTP endpoint with the ID 0.
+```bash
+curl -X DELETE http://localhost:8080/people/delete?id=0
+
+### Output: deleted
+```
```
Comment (sagar-jadhav) [inline http_srv.md:131]: Add `deletePerson` code snippet here as follows:

```
func deletePerson(w http.ResponseWriter, r *http.Request) {
	if r.Method == "DELETE" {
		id := r.FormValue("id")

		idInt, err := strconv.Atoi(id)
		if err != nil {
			log.Println(err)
		}

		_, ok := people[idInt]
		if !ok {
			w.WriteHeader(http.StatusNotFound)
			return
		}

		delete(people, idInt)
		w.WriteHeader(http.StatusOK)
		w.Write([]byte("deleted"))
	} else {
		w.Write([]byte("bad request"))
	}
}
```

File: http_srv.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+The `people` map should now be empty.
+Check this by calling the list function again
+
+```bash
+curl http://localhost:8080/people
+
+### Output: {}
+```
```
Comment (sagar-jadhav) [inline http_srv.md:166]: Add a link to `source` code at the end and also link to go back.

**Review by sagar-jadhav — CHANGES_REQUESTED** [review]

**Comment by sagar-jadhav** [comment]
@davidkroell Excellent Documentation 👍 

I suggested some minor changes please have a look. 

Please also update link in **README** as `./http_srv.html`

**Review by davidkroell — COMMENTED** [review]
Now updated as requested

**Review by sagar-jadhav — APPROVED** [review]

**Pull request merged**